provider/pkg/provider/provider.go

```diff
@@ -503,24 +505,17 @@ func (p *googleCloudProvider) Create(ctx context.Context, req *rpc.CreateRequest
 	body := p.prepareAPIInputs(inputs, nil, res.Create.SDKProperties)
 
 	var op map[string]interface{}
-	if res.AssetUpload {
-		var content []byte
-		source := inputs["source"]
-		if source.IsAsset() {
-			content, err = source.AssetValue().Bytes()
-		} else if source.IsArchive() {
-			content, err = source.ArchiveValue().Bytes(resource.ZIPArchive)
-		}
-		if err != nil {
-			return nil, err
-		}
+	var contentType string
+	if val, hasContentType := inputs["contentType"]; hasContentType {
+		contentType = val.StringValue()
+	}
 
-		op, err = p.client.UploadWithTimeout(res.Create.Verb, uri, body, content, 0)
-		if err != nil {
-			return nil, fmt.Errorf("error sending upload request: %s: %q %+v %d", err, uri, inputs.Mappable(), len(content))
-		}
+	if res.AssetUpload {
+		op, err = p.handleAssetUpload(uri, &res, inputs, body)
+	} else if contentType == "multipart/form-data" && len(res.FormDataUpload.FormFields) > 0 {
```

Comment:
Can you use `needsMultiPartFormdataContentType` here as well?

Reply:
Sorry missed this. Fixed.